launcher/ui/pagedialog/PageDialog.cpp

```diff
@@ -29,34 +29,40 @@
 PageDialog::PageDialog(BasePageProvider* pageProvider, QString defaultId, QWidget* parent) : QDialog(parent)
 {
     setWindowTitle(pageProvider->dialogTitle());
-    m_container = new PageContainer(pageProvider, defaultId, this);
+    m_container = new PageContainer(pageProvider, std::move(defaultId), this);
 
-    QVBoxLayout* mainLayout = new QVBoxLayout;
+    QVBoxLayout* mainLayout = new QVBoxLayout(this);
```

Comment:
I don't think you need the explicit parent relation if we setLayout with this object